test/OpenTelemetry.AutoInstrumentation.Tests/Configurations/ConfigurationTests.cs

```diff
@@ -33,106 +33,66 @@ private enum TestEnum
     }
 
     [Fact]
-    public void ParseEnabledEnumList_Default()
+    public void ParseEnabledEnumList_Default_Enabled()
     {
         var source = new Configuration(new NameValueConfigurationSource(new NameValueCollection()));
 
         var list = source.ParseEnabledEnumList<TestEnum>(
-            enabledConfiguration: "TEST_ENABLED_VALUES",
-            disabledConfiguration: "TEST_DISABLED_VALUES",
-            error: "Invalid enum value: {0}");
+            enabledByDefault: true,
+            enabledConfigurationTemplate: "TEST_CONFIGURATION_{0}_ENABLED");
 
         list.Should().Equal(TestEnum.Test1, TestEnum.Test2, TestEnum.Test3);
     }
 
     [Fact]
-    public void ParseEnabledEnumList_Enabled()
+    public void ParseEnabledEnumList_Default_Disabled()
     {
-        var source = new Configuration(new NameValueConfigurationSource(new NameValueCollection()
-        {
-            { "TEST_ENABLED_VALUES", "Test1,Test3" }
-        }));
+        var source = new Configuration(new NameValueConfigurationSource(new NameValueCollection()));
 
         var list = source.ParseEnabledEnumList<TestEnum>(
-            enabledConfiguration: "TEST_ENABLED_VALUES",
-            disabledConfiguration: "TEST_DISABLED_VALUES",
-            error: "Invalid enum value: {0}");
+            enabledByDefault: false,
+            enabledConfigurationTemplate: "TEST_CONFIGURATION_{0}_ENABLED");
 
-        list.Should().Equal(TestEnum.Test1, TestEnum.Test3);
+        list.Should().BeEmpty();
     }
 
     [Fact]
-    public void ParseEnabledEnumList_Disabled()
+    public void ParseEnabledEnumList_SelectivelyEnabled()
     {
-        var source = new Configuration(new NameValueConfigurationSource(new NameValueCollection()
+        var source = new Configuration(new NameValueConfigurationSource(new NameValueCollection
         {
-            { "TEST_DISABLED_VALUES", "Test2" }
+            { "TEST_CONFIGURATION_Test1_ENABLED", "true" },
```

Comment:
Just a random example here.

Can we do keys in all caps? this format is very unfriendly in UX wise, specially in Linux where a key's case sensitivity matters.

Reply:
I agree. I have forgotten to mention it in the issue. This is what Java does and also something similar was proposed here https://github.com/open-telemetry/opentelemetry-specification/issues/3025#issuecomment-1342834029

However, I think it would be better to tackle it in a separate PR.

Reply:
#2094